astroplan/core.py

```diff
@@ -649,10 +734,20 @@ def calc_rise(self, time, target, which='nearest', horizon=0*u.degree):
                 return previous_rise
 
         if which == 'nearest':
-            if abs(time - previous_rise) < abs(time - next_rise):
-                return previous_rise
+            if _target_is_vector(target):
```

Comment:
There is one fringe case in which this will fail: if the user passes in a list of targets, and one of them never rises. Right now this will die I think because at the end you pass all return times in to the `Time` constructor. I think we probably want to "soft fail" in keeping with our conventions elsewhere. Not sure what the best solution is -- one (slightly bad) option is to instead return a list of `Time`'s with a `nan` wedged in if the rise time fails for a given target...@eteq, thoughts?